File: Api/ApprovalRequests/Models/ApprovalRequest.cs
using Api.Employees.Models;
using Api.LeaveRequests.Models;

namespace Api.ApprovalRequests.Models;

public record ApprovalRequest
{
    public int Id { get; init; }

    public required EmployeeBrief Approver { get; init; }

    public required LeaveRequest LeaveRequest { get; init; }

    public required string Status { get; init; }

    public string? Comment { get; init; }
}

File: client/src/approvalRequests/models/ApprovalRequest.ts
import Employee from "src/employees/models/Employee";
import LeaveRequest from "src/leaveRequests/models/LeaveRequest";

type ApprovalRequest = {
  id: number;
  status: string;
  comment: string;
  approver: Employee;
  leaveRequest: LeaveRequest;
}

export default ApprovalRequest;